session/session.go

```diff
@@ -2544,6 +2607,9 @@ func (s *session) Close() {
 			logutil.BgLogger().Error("release table lock failed", zap.Uint64("conn", s.sessionVars.ConnectionID))
 		}
 	}
+	if s.hasAdvisoryLocks() {
```

Comment:
Is `hasAdvisoryLocks()` needed, can we not just call s.ReleaseAllAdvisoryLocks() regardless?